# LARGE RESIDENTIAL PROJECT PROPOSAL
With this project, we are trying to propose a residential multi-use large-size project at KUMBALAGODU.  The selected site has about 8% vegetation with coconut trees and deciduous trees. The site is located along the Eastern edge of Mysore Road with a road width of 45 meters and on the northwest side is a railway track at a distance of 100 m from the site. A water body is also visible on the Southwest side of the site. 

There is no existing structures or sewage lines inside the site nearest sewage lines is outside the side boundary on the eastern side of the Mysore Road.
![image](https://i.imgur.com/uv4rDRo.png)

# Project outline

- Proposal of a project

- Study current market trends in real estate 

- Real estate data analysis

- Installing and importing libraries

- Accessing data from an websites

- Extracting data and compiling into a data frame

- Storing data into an excel

# LOCATION
## The site is located 17.5 km from majestic 

### Nearest office spaces are:

● GLOBAL VILLAGE TECH PARK 6.8KM 

● GRAM PANCHAYAT OFFICE KUMBALGODU-1.3KM 

● POST OFFICE-1.6KM 

● BBMP OFFICE-4.8KM

● SUB REGISTRAR OFFICE-5.4KM 

● FEATHERLITE OFFICE SYSTEMS-1.7KM 

● PEPSICO-2.3KM 

● AMRUT DISTILLERIES-ACROSS THE ROAD

### NEAREST BUS STOP(TRAVEL TIME) 

● RAMOHALLI CROSS-450M 

● CS JUGGAT PHARMA-280 M

### NEAREST METRO STATION

● MYSORE ROAD METRO STATION(existing)-11.7KM

● CHELLAGHTA METRO DEPOT(proposed)-1.1KM

### NEAREST ECONOMIC HUB

● RR NAGAR-11KM

● KENGERI-6.1KM

# REGULATORY
- DEVELOPMENT AUTHORITY : BANGALORE DEVELOPMENT AUTHORITY.
     PLANNING DISTRICT : 322,
     KUMBALAGODU
- CDP TO BE CONSIDERED : EXISTING LAND USE - RING 3
     REVISED MASTER PLAN , 2015 , BENGALURU
![image](https://i.imgur.com/HNgceyN.png)

● SITE AREA : 11.73 Acres = 510917.51 SQ.FT
● ROAD WIDTH : 45 M
● PROPOSED LAND USE : RESIDENTIAL MIXED
● Open area +Civic amenities : 10% + 5%
● FSI : 3.25
● GROUND COVERAGE : 50%

# REGULATIONS
![image](https://i.imgur.com/LnwdYKC.png)
![image](https://i.imgur.com/s7YqykC.png)
![image](https://i.imgur.com/5z1pzsQ.png)
![image](https://i.imgur.com/L0lQbnH.png)

# MARKET SURVEY - By web scraping

# **Installs & Imports**

In [1]:
import urllib.request as urllib2
import csv
import requests
import re
from bs4 import BeautifulSoup
import pandas as pd
import jovian
# Execute this to save new versions of the notebook
jovian.commit(project='Scraping Real estate website and analysis using Python')

<IPython.core.display.Javascript object>

[jovian] Updating notebook "rakesh-rajagopalachary/scraping-real-estate-website-and-analysis-using-python" on https://jovian.ai/
[jovian] Committed successfully! https://jovian.ai/rakesh-rajagopalachary/scraping-real-estate-website-and-analysis-using-python


'https://jovian.ai/rakesh-rajagopalachary/scraping-real-estate-website-and-analysis-using-python'

In [2]:
def response (url):
    response = requests.get(url)
    response1 = response.status_code
    if response1 == 200:
        print('Pass')
    if response1 != 200:
        return response1

In [3]:
def doc (url):
    response = requests.get(url)
    page_contents = response.text
    with open('text.html', 'w', encoding="utf-8") as file:
        file.write(page_contents)
#Reading the page contents from a html file    
    with open('text.html', 'r') as f:
        html_source = f.read()
# Converting the page to  Beautiful soup document using html.parser    
    doc = BeautifulSoup(html_source, 'html.parser')
    return doc

In [4]:
def soup (url):
    #Request to pass the page with user agent which is going to access the website as a browser not as a automated scraper.
    request = urllib2.Request(url, headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36'})
    page = urllib2.urlopen(request)
    #For beautiful soup 'html.parser' is the best parser out of many parsers avilable
    soup = BeautifulSoup(page, 'html.parser')
    return soup


In [5]:
def df (x):
    df = pd.DataFrame(x)
    return df

In [109]:
def loop(x):
    banthu =[result.find(x) for result in list_items]
    return banthu
    

In [7]:
def loop_all(x):
    banthu =[result.find_all(x).get_text() for result in list_items]
    return banthu

# MAGIC BRICKS
![image](https://i.imgur.com/BsKiH2M.jpg)

In [8]:
url = "https://www.magicbricks.com/property-for-sale/residential-real-estate?&proptype=Multistorey-Apartment,Builder-Floor-Apartment,Penthouse,Studio-Apartment,Service-Apartment,Residential-House,Villa,Residential-Plot&cityName=Bangalore&bedrooms=11700,11703,11704,11701,11702&mbTrackSrc=tabChange&page=1&category=S"

In [9]:
response (url)

Pass


In [10]:
Soup = soup (url)

In [11]:
#result item (starting point)
list_items = Soup.find_all('div',{'class':'mb-srp__list'})

In [12]:
#Number of photos availiable
list_items[0].find('span',{'class':'mb-srp__card__photo__fig--count'}).get_text() 

'12+'

In [13]:
#Ad title
list_items[0].find('h2',{'class':'mb-srp__card--title'}).get_text()

'2 BHK Flat for Sale in Hennur Main Road '

In [14]:
#Society name
list_items[0].find('a',{'class':'mb-srp__card__society--name'}).get_text()

'Purva Promenade'

In [15]:
(list_items[0].find_all('div',{'class':'mb-srp__card__summary--value'})[3]).get_text()

'Unfurnished'

In [16]:
#Description
list_items[0].find('div',{'class':'mb-srp__card--desc--text'}).get_text()

'This property embraces 19.37 acres of land distinguished by spacious residential properties and exquisitely landscaped earth. Just few minutes off Hennur Road, these exquisitely styled Apartments have areas dedicated to a magnificent wave pool, Koi pond, snorkeling zone, a fully loaded gym with steam & sauna. Apartments in this location have seen a huge rise in demand due to the fast-growing infrastructure and would be an ideal destination to invest in for a quality living.'

In [17]:
#Cost
list_items[0].find('div',{'class':'mb-srp__card__price--amount'}).get_text() 

'₹99 Lac'

In [18]:
#Cost per sqft
list_items[0].find('div',{'class':'mb-srp__card__price--size'}).get_text()

'₹8036 per sqft'

In [19]:
# creating list
Number_photos =[result.find('span',{'class':'mb-srp__card__photo__fig--count'}).get_text() for result in list_items]
Title = [result.find('h2',{'class':'mb-srp__card--title'}).get_text() for result in list_items]
Society = [result.find('a',{'class':'mb-srp__card__society--name'}).get_text() for result in list_items]
#built = [result.find('div',{'class':'mb-srp__card__summary--value'}).get_text() for result in list_items]
Description = [result.find('div',{'class':'mb-srp__card--desc--text'}).get_text() for result in list_items]
Cost = [result.find('div',{'class':'mb-srp__card__price--amount'}).get_text() for result in list_items]
Cost_sft = [result.find('div',{'class':'mb-srp__card__price--size'}).get_text() for result in list_items]
Super_built = [(result.find_all('div',{'class':'mb-srp__card__summary--value'})[0]).get_text() for result in list_items]
Property = [(result.find_all('div',{'class':'mb-srp__card__summary--value'})[1]).get_text()for result in list_items]
Furniture = [(result.find_all('div',{'class':'mb-srp__card__summary--value'})[3]).get_text() for result in list_items]
Parking = [(result.find_all('div',{'class':'mb-srp__card__summary--value'})[4]).get_text() for result in list_items]


In [20]:
Magic_Brick = df({'Title':Title, 'Project': Society, 'Cost': Cost, 'Cost Per Sqft': Cost_sft, 'Data':Description})

In [21]:
Magic_Brick

Title  \
0            2 BHK Flat for Sale in Hennur Main Road    
1               2 BHK Flat for Sale in Sarjapur Road    
2            2 BHK Flat for Sale in Kadubeesanahalli    
3                  2 BHK Flat for Sale in whitefield    
4                     1 BHK Flat for Sale in Varthur    
5                   2 BHK Flat for Sale in J P Nagar    
6       2 BHK Flat for Sale in Thanisandra Main Road    
7               3 BHK Flat for Sale in Sarjapur Road    
8             2 BHK Flat for Sale in Old Madras Road    
9              2 BHK Flat for Sale in Sahakara Nagar    
10                   3 BHK Flat for Sale in RT Nagar    
11           3 BHK Flat for Sale in Hennur Main Road    
12                2 BHK Flat for Sale in Mysore Road    
13      3 BHK Flat for Sale in Thanisandra Main Road    
14                     3 BHK Flat for Sale in Jakkur    
15                     2 BHK Flat for Sale in Kogilu    
16               1 BHK Flat for Sale in Bommanahalli    
17                 2 BHK Flat for Sale in Chandapura    
18              2 BHK Flat for Sale in Sarjapur Road    
19               2 BHK Flat for Sale in Bommanahalli    
20              1 BHK Flat for Sale in Sarjapur Road    
21  3 BHK Flat for Sale in Hosa Road, off Hosur Road    
22  2 BHK Flat for Sale in Kaggadasapura, Indira n...   
23                 2 BHK Flat for Sale in Chandapura    
24  2 BHK Flat for Sale in Jp Nagar Phase 7, J P N...   
25              3 BHK Flat for Sale in Sarjapur Road    
26        1 BHK Flat for Sale in Bidadi, Mysore Road    
27            2 BHK Flat for Sale in Electronic City    
28              4 BHK Flat for Sale in Akshaya Nagar    
29                3 BHK Flat for Sale in Thanisandra    

                            Project       Cost    Cost Per Sqft  \
0                   Purva Promenade    ₹99 Lac   ₹8036 per sqft   
1                 Bavisha Gardeniaa  ₹39.3 Lac   ₹4500 per sqft   
2                    Bhuvi By Amsha   ₹1.33 Cr   ₹9541 per sqft   
3                      Sai Sunshine  ₹64.7 Lac   ₹5460 per sqft   
4                 Candeur Signature  ₹58.2 Lac   ₹9403 per sqft   
5                   Bhoomika Elites  ₹51.2 Lac   ₹4600 per sqft   
6              Visalakshi Prakruthi    ₹61 Lac   ₹5985 per sqft   
7                     Mana Capitol   ₹98.9 Lac   ₹6800 per sqft   
8                        AV Sunrise  ₹58.7 Lac   ₹4600 per sqft   
9                    Fortuna Acacia  ₹95.3 Lac   ₹8667 per sqft   
10                 HM Tropical Tree   ₹2.58 Cr  ₹12000 per sqft   
11                   Kumar Prospera   ₹1.11 Cr   ₹7604 per sqft   
12                     Meda Greens   ₹73.2 Lac   ₹5999 per sqft   
13                     Aparna Maple    ₹98 Lac   ₹6980 per sqft   
14                  Bollineni Astra  ₹95.2 Lac   ₹6242 per sqft   
15             Adarsh Green Phase 2  ₹49.4 Lac   ₹5168 per sqft   
16               Mahaveer Turquoise  ₹28.9 Lac   ₹5468 per sqft   
17                   Global Celesta  ₹32.4 Lac   ₹3523 per sqft   
18              Abhee Silicon Shine  ₹73.4 Lac   ₹6067 per sqft   
19                   SB Temple Tree   ₹1.09 Cr   ₹7539 per sqft   
20                    GRC Subhiksha  ₹45.6 Lac   ₹6612 per sqft   
21                        GR Sitara    ₹90 Lac   ₹5489 per sqft   
22            Mathapathi Brundaavan  ₹77.9 Lac   ₹6180 per sqft   
23               Signature Splendor  ₹40.9 Lac   ₹3952 per sqft   
24                   Rain bow Haven  ₹65.7 Lac   ₹5763 per sqft   
25  The Prestige City Meridian Park   ₹1.14 Cr   ₹6847 per sqft   
26                   Sattva Ashraya  ₹24.7 Lac   ₹4748 per sqft   
27                ICON Happy Living    ₹36 Lac   ₹5341 per sqft   
28                    Jampa Luxuria   ₹2.97 Cr   ₹7250 per sqft   
29                    DSMAX Skycity   ₹1.21 Cr   ₹6000 per sqft   

                                                 Data  
0   This property embraces 19.37 acres of land dis...  
1   Bavisha Gardenia is a new Creation defining ur...  
2   A Home Where Everything is a B

# Common floors 
![image](https://i.imgur.com/IUttEWU.png)

In [22]:
url='https://www.commonfloor.com/listing-search?city=Bangalore&cg=Bangalore%20division&iscg=&search_intent=sale&polygon=1&page=1&page_size=30'

In [23]:
response (url)

Pass


In [24]:
Soup = soup (url)

In [25]:
#result item (starting point)
list_items = Soup.find_all('div',{'class':'snb-tile-info'})

In [26]:
#Title
list_items[0].find_all('a')[1].get_text() 

'Plot for Sale in Jigini Anekal Road'

In [27]:
#Project
list_items[0].find_all('a')[2].get_text() 

'Jigini Anekal Road'

In [28]:
#Area
list_items[0].find('small',{'class':'smtext'}).get_text() 

' (111.48 sq.m)'

In [29]:
#Area
list_items[0].find('span').get_text().replace('\n','')

' 23.99 L         '

In [30]:
list_items[0].find('div',{'class':'p_section'}).find('span',{'class':'s_p'}).get_text().replace('\n','')

' 23.99 L         '

In [31]:
list_items[0].find('div',{'class':'s_psft'}).get_text() 


' 1999 per sq.ft. Plot Area'

In [34]:
# creating list
Title =[result.find_all('a')[1].get_text() for result in list_items]
Project = [result.find_all('a')[2].get_text() for result in list_items]
#Area = [result.find('small',{'class':'smtext'}).get_text() for result in list_items]
vary = [result.find('span').get_text().replace('\n','') for result in list_items]
Cost = [result.find('div',{'class':'p_section'}).find('span',{'class':'s_p'}).get_text().replace('\n','') for result in list_items]
Cost_sft = [result.find('div',{'class':'s_psft'}).get_text() for result in list_items]

In [35]:
Common = df({'Title': Title, 'Project':Project, 'Cost':Cost, 'Cost per Sqft':Cost_sft, 'Data':vary})

In [36]:
Common

Title  \
0                 Plot for Sale in Jigini Anekal Road   
1   Not Furnished 2 BHK Apartment for Sale in Kann...   
2    Semi Furnished 3BHK Apartment for Sale in Hebbal   
3   Not Furnished 1 BHK Apartment for Sale in Bela...   
4                        Plot for Sale in Devanahalli   
5                         Plot for Sale in Chandapura   
6                         Plot for Sale in Chandapura   
7   Semi Furnished 4BHK Apartment for Sale in Asho...   
8    Semi Furnished 4BHK Apartment for Sale in Ulsoor   
9   Semi Furnished 3BHK Apartment for Sale in Kana...   
10  Semi Furnished 3BHK Apartment for Sale in Cunn...   
11  Semi Furnished 3BHK Apartment for Sale in Bell...   
12  Semi Furnished 4BHK Apartment for Sale in Mall...   
13  Semi Furnished 4BHK Apartment for Sale in Vasa...   
14  Semi Furnished 4BHK Villa for Sale in Old Madr...   
15  Semi Furnished 4BHK Apartment for Sale in Lalb...   
16  Semi Furnished 2.5BHK Apartment for Sale in Be...   
17  Semi Furnished 4BHK Apartment for Sale in Shan...   
18  Semi Furnished 4BHK Apartment for Sale in Lalb...   
19    Semi Furnished 4BHK Villa for Sale in Yelahanka   
20   Semi Furnished 3BHK Apartment for Sale in Hebbal   
21  Semi Furnished 4BHK Apartment for Sale in Mall...   
22  Semi Furnished 3BHK Apartment for Sale in Rich...   
23  Semi Furnished 3BHK Apartment for Sale in Doll...   
24  Semi Furnished 4BHK Villa for Sale in Bannergh...   
25          4BHK Apartment for Sale in Dollars Colony   
26   Semi Furnished 4BHK Villa for Sale in Whitefield   
27  Semi Furnished 3BHK Villa for Sale in Bannergh...   
28   Semi Furnished 4BHK Apartment for Sale in Jakkur   
29    Semi Furnished 3BHK Villa for Sale in Yelahanka   

                                              Project  \
0                                  Jigini Anekal Road   
1                        Dwarakamai Apex 3 Apartments   
2                                      Brigade Laguna   
3                                Green Prime Manshion   
4                                         Devanahalli   
5                                       JR Green Park   
6                                       JR Green Park   
7                          Prestige Kingfisher Towers   
8                                  Prestige Hermitage   
9   Total Environment The Magic Faraway Tree Phase II   
10                                Prestige Kenilworth   
11                                       Purva Zenium   
12                                    Mantri Centrium   
13                                    Legacy Cataleya   
14                                Sattva Kings Domain   
15                                 Purva Orient Grand   
16                                       Purva Zenium   
17                                     Mantri Blossom   
18                                 Purva Orient Grand   
19                              True Blue Napa Valley   
20                                       SNN Clermont   
21                                     Sattva Luxuria   
22                                  White Lotus Ohana   
23                                 Prestige Fairfield   
24                              Purva Smiling Willows   
25                                 Prestige Fairfield   
26                          NVT Stopping by the Woods   
27                              Purva Smiling Willows   
28                                     Legacy Cirocco   
29                              True Blue Napa Valley   

                                Cost                  Cost per Sqft  \
0                   23.99 L               1999 per sq.ft. Plot Area   
1                   57.75 L             6875 per sq.ft. Carpet Area   
2                   1.50 Cr             9659 per sq.ft. Carpet Area   
3                   17.12 L             4127 per sq.ft. Carpet Area   
4             Call For Price                   per sq.ft. Plot Area   
5                      48 L               4000 per sq.ft. Plot Area   
6              

# HOUSING.COM
![image](https://i.imgur.com/FR954nY.png)

In [37]:
url='https://housing.com/in/buy/searches/P38f9yfbk7p3m2h1f'

In [38]:
response (url)

Pass


In [39]:
Soup = soup (url)

In [40]:
#result item (starting point)
list_items = Soup.find_all('article',{'class':'css-1nr7r9e'})

In [41]:
len(list_items)

20

In [42]:
#Title
list_items[0].find('div',{'class':'css-2rx1iy'}).find('a').get_text() 

'Sai Prakruthi Gardeniaa'

In [43]:
#Cost
list_items[0].find('div',{'class':'css-18rodr0'}).get_text()

'₹62 L - 93 L'

In [44]:
#Property
list_items[0].find('h3',{'class':'css-197fqpq'}).get_text()

'Residential Plots for sale in Kogilu, Bangalore'

In [45]:
#Address
list_items[0].find('div',{'class':'css-4in8py'}).get_text()

'Kogilu,North Bangalore,Bangalore'

In [46]:
#Cost per sqft
list_items[0].find_all('div',{'class':'css-4z3njv'})[1].get_text()

'₹6.20 K/sq.ft'

In [47]:
#Project
list_items[0].find('div',{'class':'css-wni7av'}).get_text()

'Sai Akshaya Ventures'

In [48]:
# creating list
Title =[result.find('div',{'class':'css-2rx1iy'}).find('a').get_text()for result in list_items]
Cost = [result.find('div',{'class':'css-18rodr0'}).get_text() for result in list_items]
Project = [result.find('div',{'class':'css-wni7av'}).get_text()for result in list_items]
Address = [result.find('div',{'class':'css-4in8py'}).get_text()for result in list_items]
#Status = [result.find('div',{'class':'css-4in8py'}).get_text() for result in list_items]
Cost_sft = [result.find_all('div',{'class':'css-4z3njv'})[1].get_text()for result in list_items]

In [49]:
Housing = df({'Title': Title, 'Project':Project, 'Cost':Cost, 'Cost per Sqft':Cost_sft, 'Data':Address})

In [50]:
Housing

Title  \
0                             Sai Prakruthi Gardeniaa   
1                                    Godrej Splendour   
2                                    SRI RAMA KUTEERA   
3                                      SB Eco Springs   
4                                             Centreo   
5                                   Sammys Palm Hills   
6                        Manyata Earthsong By Manyata   
7                        Gallium At Brigade El Dorado   
8                                        Century Eden   
9                       Prestige Finsbury Park Regent   
10                        Prestige Finsbury Park Hyde   
11                       Bavisha Urban Homes Phase II   
12                                Provident Equinox 2   
13                             GM Global Techies Town   
14                                    Bollineni Astra   
15                               Prestige Jindal City   
16       3 BHK Flat for sale in HBR Layout, Bangalore   
17  3 BHK Flat for sale in Electronic City, Bangalore   
18   3 BHK Flat for sale in Lingarajapuram, Bangalore   
19         2 BHK Flat for sale in Horamavu, Bangalore   

                    Project                Cost     Cost per Sqft  \
0      Sai Akshaya Ventures        ₹62 L - 93 L     ₹6.20 K/sq.ft   
1   Godrej Properties Ltd.   ₹41.99 L - 89.99 L     ₹5.90 K/sq.ft   
2            Charigan Group    ₹20.4 L - 40.8 L     ₹1.70 K/sq.ft   
3           S B Urbanscapes        ₹42 L - 84 L  Price on request   
4         Property Junction  ₹1.38 Cr - 2.04 Cr     ₹8.50 K/sq.ft   
5         Property Junction     ₹17.63 L - 68 L     ₹1.70 K/sq.ft   
6          Space Properties  ₹48.75 L - 1.63 Cr     ₹3.25 K/sq.ft   
7         Property Junction     ₹36 L - 46.08 L     ₹4.04 K/sq.ft   
8         Property Junction  ₹77.35 L - 1.14 Cr     ₹3.40 K/sq.ft   
9         Property Junction     ₹70.01 L - 84 L  Price on request   
10            Dilip Kumar R   ₹55.0 L - 57.53 L  Price on request   
11     Bhavisha Properties     Price on Request  Price on request   
12        BricksBee Pvt Ltd    ₹54.5 L - 65.3 L     ₹8.75 K/sq.ft   
13        BricksBee Pvt Ltd   ₹59.06 L - 1.1 Cr  Price on request   
14       GR Reality Pvt Ltd   ₹76.5 L - 1.59 Cr     ₹6.38 K/sq.ft   
15        Property Junction     ₹72 L - 1.24 Cr  Price on request   
16                 Srinidhi            ₹1.12 Cr     ₹7.52 K/sq.ft   
17                    Vivek             ₹99.9 L     ₹6.42 K/sq.ft   
18          Owais O Vahedna             ₹3.5 Cr    ₹14.00 K/sq.ft   
19       Pioneer Associates            ₹1.29 Cr     ₹9.42 K/sq.ft   

                                                 Data  
0                    Kogilu,North Bangalore,Bangalore  
1                Whitefield,Bangalore East ,Bangalore  
2           Chikkaballapura,North Bangalore,Bangalore  
3              Budigere Cross,Budigere Road,Bangalore  
4                          Kanakapura Road, Bangalore  
5                              Devanahalli, Bengaluru  
6                 Uganavadi,North Bangalore,Bangalore  
7                 Aerospace Park, Bagaluru, Bengaluru  
8             Bisuvanahalli,North Bangalore,Bangalore  
9                                  Bagalur, Bengaluru  
10                Aerospace Park, Bagaluru, Bengaluru  
11                   Sompura Gate, Sarjapur Main Road  
12                             Mysore Road, Bengaluru  
13              Bommasandra,South Bangalore,Bangalore  
14                   Kogilu,North Bangalore,Bangalore  
15                             Tumkur Road, Bengaluru  
16  Casagrand Orlena, Ashwath Nagar, HBR Layout,Ba...  
17  SV Grandur, Kammasandra, Electronic City,Banga...  
18  Ambience Downtown, Joseph Reddy Layout, Lingar...  
19  Prestige Gulmohar, Jayanti Nagar, Horamavu,Ban...

# No Broker
![image](https://i.imgur.com/AiadLc5.jpg)

In [79]:
url= 'https://www.nobroker.in/property/sale/bangalore/multiple?searchParam=W3sibGF0IjoxMi44OTk2Njc2LCJsb24iOjc3LjQ4MjY4MzcsInBsYWNlSWQiOiJDaElKdDlJVHlUQV9yanNSb3N5SnpFN0p1LWsiLCJwbGFjZU5hbWUiOiJLZW5nZXJpIn0seyJsYXQiOjEyLjg3OTQwMDgsImxvbiI6NzcuNDQ1MTMxNjk5OTk5OTksInBsYWNlSWQiOiJDaElKMFE3Z1JaazRyanNSVE5pUkRGdEd0eU0iLCJwbGFjZU5hbWUiOiJLdW1iYWxnb2R1In1d&radius=2.0&city=bangalore&locality[0]=Kengeri,&locality[1]=Kumbalgodu&propType=AP&type=RK1,BHK1,BHK2,BHK3,BHK4,BHK4PLUS'

In [80]:
response (url)

Pass


In [81]:
Soup = soup (url)

In [82]:
#result item (starting point)
list_items = Soup.find_all('span',{'itemprop':'itemListElement'})

In [83]:
len(list_items)

50

In [84]:
#Title
list_items[0].find('span',{'class':'overflow-hidden overflow-ellipsis whitespace-nowrap max-w-80pe po:max-w-full'}).get_text() 

'2 BHK Flat  For Sale  In Vaishnodevi Excelsior In Kumbalgodu'

In [85]:
#Project
list_items[0].find('span',{'class':'overflow-hidden overflow-ellipsis whitespace-nowrap max-w-80pe po:max-w-full'}).get_text() 

'2 BHK Flat  For Sale  In Vaishnodevi Excelsior In Kumbalgodu'

In [101]:
#Cost
list_items[0].find('div',{'class':'p-1.5p flex border-b border-b-solid border-cardbordercolor tp:py-1p tp:px-1.5p tp:border-b-0'}).get_text()

'Not under loan₹ 1,040 sqftBuiltup₹21,206/MonthEstimated EMI₹37 Lacs₹3,558 per sq.ft.'

In [147]:
#Cost
list_items[0].find('div',{'class':'p-1.5p flex border-b border-b-solid border-cardbordercolor tp:py-1p tp:px-1.5p tp:border-b-0'}).get_text().split('EMI')[1].split('Lacs')[0]+str('Lacs')

'₹37 Lacs'

In [132]:
#Cost per sft
list_items[0].find('div',{'class':'p-1.5p flex border-b border-b-solid border-cardbordercolor tp:py-1p tp:px-1.5p tp:border-b-0'}).get_text().split('Lacs')[1]

'₹3,558 per sq.ft.'

In [108]:
#Data
list_items[0].find('div',{'class':'mt-0.5p overflow-hidden overflow-ellipsis whitespace-nowrap max-w-70 text-gray-light leading-4 po:mb-0.1p po:max-w-95'}).get_text()

'opp to christ university'

In [149]:
# creating list
Title =[result.find('span',{'class':'overflow-hidden overflow-ellipsis whitespace-nowrap max-w-80pe po:max-w-full'}).get_text()for result in list_items]
Cost = [result.find('div',{'class':'p-1.5p flex border-b border-b-solid border-cardbordercolor tp:py-1p tp:px-1.5p tp:border-b-0'}).get_text().split('EMI')[1].split('Lacs')[0]+str('Lacs')for result in list_items]
Address = [result.find('div',{'class':'mt-0.5p overflow-hidden overflow-ellipsis whitespace-nowrap max-w-70 text-gray-light leading-4 po:mb-0.1p po:max-w-95'}).get_text()for result in list_items]
Project = [result.find('span',{'class':'overflow-hidden overflow-ellipsis whitespace-nowrap max-w-80pe po:max-w-full'}).get_text() for result in list_items]
Cost_sft = [result.find('div',{'class':'p-1.5p flex border-b border-b-solid border-cardbordercolor tp:py-1p tp:px-1.5p tp:border-b-0'}).get_text().split('EMI')[1]for result in list_items]

In [153]:
No_Broker = df({'Title': Title, 'Project':Project, 'Cost':Cost, 'Cost per Sqft':Cost_sft, 'Data':Address})

In [154]:
No_Broker

Title  \
0   2 BHK Flat  For Sale  In Vaishnodevi Excelsior...   
1   2 BHK Flat  For Sale  In Vaishnodevi Excelsior...   
2   2 BHK Flat  For Sale  In Gopalan Olympia, Kumb...   
3   2 BHK Flat  For Sale  In Gopalan Olympia, Kumb...   
4                4+ BHK Flat  For Sale  In Kumbalgodu   
5                4+ BHK Flat  For Sale  In Kumbalgodu   
6   2 BHK Flat  For Sale  In Vertue Excelsior  In ...   
7   2 BHK Flat  For Sale  In Vertue Excelsior  In ...   
8   2 BHK Flat  For Sale  In Oraiyaan Excelsior In...   
9   2 BHK Flat  For Sale  In Oraiyaan Excelsior In...   
10  1 BHK Flat  For Sale  In Sowparnika Pranathi I...   
11  1 BHK Flat  For Sale  In Sowparnika Pranathi I...   
12  2 BHK Flat  For Sale  In Sowparnika Pranathi I...   
13  2 BHK Flat  For Sale  In Sowparnika Pranathi I...   
14  3 BHK Flat  For Sale  In Sowparnika Pranathi I...   
15  3 BHK Flat  For Sale  In Sowparnika Pranathi I...   
16  2 BHK Flat  For Sale  In Gopalan Olympia In Ku...   
17  2 BHK Flat  For Sale  In Gopalan Olympia In Ku...   
18  3 BHK Flat  For Sale  In Gopalan Olympia, Kumb...   
19  3 BHK Flat  For Sale  In Gopalan Olympia, Kumb...   
20  3 BHK Flat  For Sale  In Gopalan Olympia In Ku...   
21  3 BHK Flat  For Sale  In Gopalan Olympia In Ku...   
22  2 BHK Flat  For Sale  In Oraiyan Excelsior In ...   
23  2 BHK Flat  For Sale  In Oraiyan Excelsior In ...   
24  2 BHK Flat  For Sale  In Orion Exelasor In Kum...   
25  2 BHK Flat  For Sale  In Orion Exelasor In Kum...   
26  2 BHK Flat  For Sale  In Exccilor Apartment  I...   
27  2 BHK Flat  For Sale  In Exccilor Apartment  I...   
28  2 BHK Flat  For Sale  In Oraiyan Excelsior In ...   
29  2 BHK Flat  For Sale  In Oraiyan Excelsior In ...   
30  2 BHK Flat  For Sale  In Brigade Panorama In K...   
31  2 BHK Flat  For Sale  In Brigade Panorama In K...   
32  3 BHK Flat  For Sale  In Brigade Panorama, Anc...   
33  3 BHK Flat  For Sale  In Brigade Panorama, Anc...   
34  3 BHK Apartment  For Sale  In Brigade Panorama...   
35  3 BHK Apartment  For Sale  In Brigade Panorama...   
36  3 BHK Apartment  For Sale  In Brigade Panorama...   
37  3 BHK Apartment  For Sale  In Brigade Panorama...   
38  2 BHK Flat  For Sale  In Brigade Panorama, Anc...   
39  2 BHK Flat  For Sale  In Brigade Panorama, Anc...   
40  3 BHK Flat  For Sale  In Brigade Panorama, Anc...   
41  3 BHK Flat  For Sale  In Brigade Panorama, Anc...   
42  2 BHK Flat  For Sale  In Brigade Panorama, Anc...   
43  2 BHK Flat  For Sale  In Brigade Panorama, Anc...   
44  2 BHK Apartment  For Sale  In Brigade Panorama...   
45  2 BHK Apartment  For Sale  In Brigade Panorama...   
46  3 BHK Apartment  For Sale  In Brigade Panorama...   
47  3 BHK Apartment  For Sale  In Brigade Panorama...   
48  3 BHK Apartment  For Sale  In Brigade Panorama...   
49  3 BHK Apartment  For Sale  In Brigade Panorama...   

                                              Project  \
0   2 BHK Flat  For Sale  In Vaishnodevi Excelsior...   
1   2 BHK Flat  For Sale  In Vaishnodevi Excelsior...   
2   2 BHK Flat  For Sale  In Gopalan Olympia, Kumb...   
3   2 BHK Flat  For Sale  In Gopalan Olympia, Kumb...   
4                4+ BHK Flat  For Sale  In Kumbalgodu   
5                4+ BHK Flat  For Sale  In Kumbalgodu   
6   2 BHK Flat  For Sale  In Vertue Excelsior  In ...   
7   2 BHK Flat  For Sale  In Vertue Excelsior  In ...   
8   2 BHK Flat  For Sale  In Oraiyaan Excelsior In...   
9   2 BHK Flat  For Sale  In Oraiyaan Excelsior In...   
10  1 BHK Flat  For Sale  In Sowparnika Pranathi I...   
11  1 BHK Flat  For Sale  In Sowparnika Pranathi I...   
12  2 BHK Flat  For Sale  In Sowparnika Pranathi I...   
13  2 BHK Flat  For Sale  In Sowparnika Pranathi I...   
14  3 BHK Flat  For Sale  In Sowparnika Pranathi I...   
15  3 BHK Flat  For Sale  In Sowparnika Pranathi I...   
16  2 BHK Flat  For Sale  In Gopalan Olympia In Ku...   
17  2 BHK Flat  For Sale  In Gopalan Olympia In Ku...   
18  3 BHK Flat  For Sale  In Gopalan Olympia, Kumb...   


In [159]:
Real_estate = pd.concat([Magic_Brick, Common, Housing, No_Broker], axis=0)

In [160]:
Real_estate

Title  \
0            2 BHK Flat for Sale in Hennur Main Road    
1               2 BHK Flat for Sale in Sarjapur Road    
2            2 BHK Flat for Sale in Kadubeesanahalli    
3                  2 BHK Flat for Sale in whitefield    
4                     1 BHK Flat for Sale in Varthur    
..                                                ...   
45  2 BHK Apartment  For Sale  In Brigade Panorama...   
46  3 BHK Apartment  For Sale  In Brigade Panorama...   
47  3 BHK Apartment  For Sale  In Brigade Panorama...   
48  3 BHK Apartment  For Sale  In Brigade Panorama...   
49  3 BHK Apartment  For Sale  In Brigade Panorama...   

                                              Project  \
0                                     Purva Promenade   
1                                   Bavisha Gardeniaa   
2                                      Bhuvi By Amsha   
3                                        Sai Sunshine   
4                                   Candeur Signature   
..                                                ...   
45  2 BHK Apartment  For Sale  In Brigade Panorama...   
46  3 BHK Apartment  For Sale  In Brigade Panorama...   
47  3 BHK Apartment  For Sale  In Brigade Panorama...   
48  3 BHK Apartment  For Sale  In Brigade Panorama...   
49  3 BHK Apartment  For Sale  In Brigade Panorama...   

                                 Cost   Cost Per Sqft  \
0                             ₹99 Lac  ₹8036 per sqft   
1                           ₹39.3 Lac  ₹4500 per sqft   
2                            ₹1.33 Cr  ₹9541 per sqft   
3                           ₹64.7 Lac  ₹5460 per sqft   
4                           ₹58.2 Lac  ₹9403 per sqft   
..                                ...             ...   
45                           ₹65 Lacs             NaN   
46  ₹1.05 Crores₹7,241 per sq.ft.Lacs             NaN   
47  ₹1.05 Crores₹7,241 per sq.ft.Lacs             NaN   
48      ₹1 Crore₹6,897 per sq.ft.Lacs             NaN   
49      ₹1 Crore₹6,897 per sq.ft.Lacs             NaN   

                                                 Data  \
0   This property embraces 19.37 acres of land dis...   
1   Bavisha Gardenia is a new Creation defining ur...   
2   A Home Where Everything is a Blessing. A premi...   
3   Sree Sai Sunshine which has Apartments at Whit...   
4   Candeur signature is emerging as the hub for l...   
..                                                ...   
45           Brigade Panorama  Survey No. 43/5, 43/30   
46  Brigade Panorama  Survey No. 43/5, 43/30 & 238...   
47  Brigade Panorama  Survey No. 43/5, 43/30 & 238...   
48  Brigade Panorama  Survey No. 43/5, 43/30 & 238...   
49  Brigade Panorama  Survey No. 43/5, 43/30 & 238...   

                    Cost per Sqft  
0                             NaN  
1                             NaN  
2                             NaN  
3                             NaN  
4                             NaN  
..                            ...  
45      ₹65 Lacs₹6,871 per sq.ft.  
46  ₹1.05 Crores₹7,241 per sq.ft.  
47  ₹1.05 Crores₹7,241 per sq.ft.  
48      ₹1 Crore₹6,897 per sq.ft.  
49      ₹1 Crore₹6,897 per sq.ft.  

[160 rows x 6 columns]

In [162]:
Real_estate.to_csv('Real_Estate.csv')

# **Future scope**
- Application of ML in this project

# **Summary**

- The Scraping was done using Python libraries such as Requests, Beautiful soup for extracting the data
- Scraping real estate data form Real Estate website.
- Parsed all the scraped data into csv file containing 160 rows and 36 columns.

# **References**

https://www.nobroker.in/

https://www.magicbricks.com/

https://housing.com/in/

In [164]:
jovian.commit(project="REAL ESTATE DATA WEB SCRAPING")

<IPython.core.display.Javascript object>

[jovian] Creating a new project "rakesh-rajagopalachary/REAL ESTATE DATA WEB SCRAPING"
[jovian] Committed successfully! https://jovian.ai/rakesh-rajagopalachary/real-estate-data-web-scraping


'https://jovian.ai/rakesh-rajagopalachary/real-estate-data-web-scraping'